associacao-brasileira-de-normas-tecnicas-ufrgs-note-initials-with-ibid.csl

```diff
@@ -1,275 +1,491 @@
-      </else-if>
       <else>
-        <text variable="title" font-weight="bold"/>
+        <names variable="author">
+          <name form="short" delimiter="; " delimiter-precedes-last="never" initialize-with=". " name-as-sort-order="all">
+            <name-part name="family" text-case="uppercase"/>
+            <name-part name="given" text-case="capitalize-first"/>
+          </name>
+          <et-al font-style="italic"/>
+          <substitute>
+            <names variable="editor"/>
+            <names variable="translator"/>
+            <choose>
+              <if type="book">
+                <!-- uso do título quando não houver autor -->
+                <text variable="title" form="short" text-case="uppercase"/>
+              </if>
+              <else>
+                <text variable="title" form="short" text-case="uppercase" quotes="false"/>
+              </else>
+            </choose>
+          </substitute>
+        </names>
       </else>
     </choose>
-    <text value=""/>
   </macro>
-  <macro name="title-short">
+  <macro name="access">
+    <!-- Macro utilizada para apresentar a URL do site utilizado e data de acesso -->
     <choose>
-      <if variable="title" match="none">
+      <if variable="DOI" match="any">
+        <!-- Quando houver DOI, utilizá-lo no lugar do link -->
+        <text term="available at" text-case="capitalize-first" prefix=" " suffix=": "/>
+        <text variable="DOI" prefix="https://doi.org/"/>
+        <group>
+          <text term="accessed" text-case="capitalize-first" prefix=". " suffix=" "/>
+          <text term="at" text-case="lowercase" suffix=": "/>
+          <text macro="accessed"/>
+        </group>
+      </if>
+      <else-if variable="URL">
         <choose>
-          <if type="interview">
-            <text term="interview"/>
+          <if type="book chapter">
+            <!-- Quando for livro, incluir a expressão e-book em itálico -->
+            <text value="E-book" font-style="italic" suffix="."/>
           </if>
-          <else-if type="manuscript speech" match="any">
-            <text variable="genre" form="short"/>
-          </else-if>
+          <else>
+            <text term="available at" text-case="capitalize-first" prefix=" " suffix=": "/>
+            <text variable="URL"/>
+            <group>
+              <!-- formato de acesso utiliza a macro accessed -->
+              <text term="accessed" text-case="capitalize-first" prefix=". " suffix=" "/>
+              <text term="at" text-case="lowercase" suffix=": "/>
+              <text macro="accessed"/>
+            </group>
+          </else>
         </choose>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="accessed">
+    <!-- Macro utilizada para a data de acesso -->
+    <date variable="accessed" suffix="." delimiter=" ">
+      <date-part name="day"/>
+      <date-part name="month" form="short"/>
+      <date-part name="year"/>
+    </date>
+  </macro>
+  <macro name="title">
```

Comment:
I don't quite understand why you're putting the `suffix=". "` into the macro for some items but into the citation/bibliography code for others. It's very hard to read and will make this error prone. Either option is fine, but it'd be good to apply that consistently.